src/quantuminspire/api.py

```diff
@@ -546,6 +646,51 @@ def get_assets(self) -> List[Dict[str, Any]]:
         ret: List[Dict[str, Any]] = self._action(['assets', 'list'])
         return ret
 
+    def get_assets_from_project(self, project_id: int) -> List[Dict[str, Any]]:
+        """ Gets the assets with its properties for a single project, given the project id.
+
+        Args:
+            project_id: The project identification number.
+
+        Returns:
+            List of assets with its properties for the project with identification project_id.
+            An empty list is returned when the project has no assets.
+
+        Raises:
+            ApiError: If the project identified by project_id does not exist.
+        """
+        try:
+            assets = self._action(['projects', 'assets', 'list'], params={'id': project_id})
+        except ErrorMessage as err_msg:
+            raise ApiError('Project with id {} does not exist!'.format(project_id)) from err_msg
+        ret: List[Dict[str, Any]] = assets
+        return ret
+
+    def get_asset_from_job(self, job_id: int) -> Dict[str, Any]:
+        """ Gets the asset data from the job, given the job_id.
+
+        Args:
+            job_id: The identification number of the job.
+
+        Raises:
+            ApiError: If the requested asset with identification from the job 'input' field does not exist.
+
+        Returns:
+            The assets with all of its properties.
+            See `get_asset` for a description of the asset properties.
+        """
+        job = self.get_job(job_id)
+        asset_url = str(job.get('input'))
+        try:
+            asset_id = int(asset_url.split('/')[-2])
+        except (ValueError, IndexError) as err_msg:
```

Comment:
In above methods, only IndexError is excepted, not ValueError. Is this a special case or should ValueError also be included above? And, since we are on it, should this maybe be extracted since it is the same-ish block over and over again?

Reply:
The ValueError is added because in this case the asset_id is an integer and has to be converted from a substring fromthe url. In the other cases the token was a string.

I thought about extracting a piece of code but I don't think we win something because except the exception framework the variables and error messages are different so we will have to put them all in parameters.